ast2vec/tests/test_repo2base.py

```diff
@@ -0,0 +1,41 @@
+import tempfile
+import unittest
+from subprocess import CalledProcessError
+
+import ast2vec.tests as tests
+from ast2vec.repo2.source import Repo2Source, Repo2SourceTransformer
+
+repo_url = 'https://github.com/src-d/ast2vec.git'  # Suppose that we will not change it :)
+
+
+class Repo2BaseTests(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        tests.setup()
+
+    def test_prepare_reponame(self):
+        reponames = [
+            'https://whatever.cite.com/cool/you.git',
+            'whatever.cite.com/cool/you.git',
+            'whatever.cite.com/cool/you\\\n',
+            'whatever.cite.com/cool/you\n',
+            'whatever.cite.com/cool/you',
+            'whatever.cite.com/cool/you.git\n',
+        ]
+        result = 'https://whatever.cite.com/cool/you.git'
+        for rn in reponames:
+            self.assertEqual(result, Repo2Source.prepare_reponame(rn))
+
+    def test_clone_repository(self):
+        r2b = Repo2Source(linguist=tests.ENRY)
+        r2b.clone_repository(repo_url)
+        # ok if no exception
+        with tempfile.TemporaryDirectory() as tmp:
+            r2b.clone_repository(repo_url, tmp)
+            # ok if no exception
+        with self.assertRaises(CalledProcessError) as context:
+            # some bad one
+            r2b.clone_repository(repo_url + '6bIUt676B6')
```

Comment:
All right, tests should assert as much as they can. At least you should check that the files exist.